tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_ceil.h

```diff
@@ -0,0 +1,41 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+#include "sfpi.h"
+#include "noc_nonblocking_api.h"
+#include <limits.h>
+
+using namespace sfpi;
+
+namespace ckernel {
+namespace sfpu {
+
+template <bool APPROXIMATION_MODE, int ITERATIONS = 8>
+inline void calculate_ceil()
+{
+    for (int d = 0; d < ITERATIONS; d++)
+    {
+        vFloat result = dst_reg[0];
+        vFloat v = result;
+        vInt tmp = float_to_int16(result); //TODO: Replace float_to_int16 to float_to_int32 once it is available
+        result= int32_to_float(tmp);
+        v_if (result < v){
+            result = result + 1;
+        }
+        v_endif;
+        v_if (v <= SHRT_MIN || v > SHRT_MAX){
```

Comment:
Hi, 
I've put this condition first.
``` 
v_if (input <= SHRT_MIN || input > SHRT_MAX) {
    result = input;
}
v_endif;
```
This is capped at SHRT_MIN and SHRT_MAX since we are using float_to_int16 conversion in order to calculate CEIL. Without the condition, TT output for values above SHRT_MAX is capped at 32767.
- Eg: For input 40788.5, the expected output is 40788.0. However, TT result is 32767.0